File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
answer = 0

def dfs(s, v, nums):
    global answer
    
    if len(s) == 3:
        n = 0
        for i in s:
            n += nums[i]
            
        for i in range(2, n):
            if n % i == 0:
                break
        else:
            answer += 1
        return
    
    for i in range(s[-1] + 1, len(nums)):
        if not v[i]:
            s.append(i)
            v[i] = True
            dfs(s, v, nums)
            s.pop()
            v[i] = False

def solution(nums):
    global answer
    visited = [False] * (len(nums) + 1)
    selected = []
    visited[0] = True
    
    for i in range(len(nums)):
        selected.append(i)
        dfs(selected, visited, nums)
        selected.pop()

    return answer

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    int t = 3; // 목표 개수
    int answer = 0;
    
    public int solution(int[] nums) {
        makeCombination(nums, 0, 0, 0);

        return answer;
    }
    
    // 조합 생성
    private void makeCombination(int[] nums, int current, int start, int count) {
        if (count == t) {
            if (!isPrime(current)) {
                answer++;
            }
            
            return;
        }
        
        for (int i = start; i < nums.length; i++) {
            makeCombination(nums, current + nums[i], i + 1, count + 1);
        }
        
        return;
    }
    
    // 소수 판별
    private boolean isPrime(int num) {
        if (num <= 1) {
            return false;
        }
        
        for (int i = 2; i <= num / 2; i++) {
            if (num % i == 0) {
                return true;
            }
        }
        
        return false;
    }
}